demo/src/components/Cart/Cart.js

```diff
@@ -5,13 +5,28 @@ import './Cart.css';
 import {setQuantity} from '../../store/StoreHelpers.js';
 import PropTypes from 'prop-types';
 
+/**
+ * Computes the price of purchasing the given quantity
+ * of all items in the cart.
+ * @param {!Object<string,{cost:number, quantity:number}>} items The items
```

Comment:
It seems like this is essentially the same value type as used in CartBase's items parameter. I'd recommend pulling it out to a typedef, rather than just declaring it one-off for each method.

https://google.github.io/styleguide/jsguide.html#jsdoc-enum-and-typedef-comments

Reply:
Refactored